tests/integration/conftest.py

```diff
@@ -350,3 +352,27 @@ def backend_metric(backend, metric_params) -> Metric:
     resource = backend.metrics.create(params=metric_params)
     yield resource
     cleanup(resource)
+
+@pytest.fixture(scope="module")
+def custom_tenant(master_api, tenant_params):
+    """
+    Fixture for getting the custom tenant.
+    """
+    resource = master_api.tenants.create(tenant_params)
+    yield resource
+    resource.delete()
+    # tenants are not deleted immediately, they are scheduled for the deletion
+    # the exists method returns ok response, even if the tenant is scheduled for deletion
+    # However, the deletion of the tenant fails, if already deleted
+    with pytest.raises(errors.ApiClientError):
```

Comment:
This check if the tenant has been deleted is not nice, but I have not found a better way.

Reply:
@DavidRajnoha I think you should wait here for some time until `resouce.delete` returns exception.

Reply:
Ok, so after chat with @DavidRajnoha we've agreed that this is best solution.